packages/base-crypto/src/VirgilCrypto.ts

```diff
@@ -280,7 +437,16 @@ export class VirgilCrypto implements ICrypto {
 
     const publicKeys = toArray(publicKey);
     validatePublicKeysArray(publicKeys);
-    const lowLevelPublicKeys = publicKeys.map(publicKey => importPublicKey(publicKey.key));
+    const lowLevelPublicKeys: LowLevelPublicKey[] = [];
+    publicKeys.forEach(({ key }) => {
+      try {
+        const lowLevelPublicKey = importPublicKey(key);
+        lowLevelPublicKeys.push(lowLevelPublicKey);
+      } catch (error) {
+        lowLevelPublicKeys.forEach(lowLevelPublicKey => lowLevelPublicKey.delete());
+        throw error;
+      }
+    });
 
     const recipientCipher = new this.foundationModules.RecipientCipher();
     const aes256Gcm = new this.foundationModules.Aes256Gcm();
```

Comment:
The signature is calculated after constructing `RecipientCipher` and is not wrapped in `try` block. The `const signature = this.calculateSignature(myData, privateKey);` line should be moved before these two lines